xbmc/addons/AddonManager.cpp

```diff
@@ -830,6 +831,7 @@ bool CAddonMgr::DisableAddon(const std::string& id)
   AddonPtr addon;
   if (GetAddon(id, addon, ADDON_UNKNOWN, false) && addon != NULL)
   {
+    CServiceBroker::GetBinaryAddonManager().DisableEvent(addon->Type(), addon->ID());
```

Comment:
Will the CBinaryAddonManager define itself what it does. In case of way changes need only the related manager updated.